scripts/encode_data.py

```diff
@@ -43,15 +48,46 @@ def embed_documents(self) -> None:
             if name == "data/links":
                 continue
 
-            thread = threading.Thread(target=self._process_document, args=(file,))
+            threading.Thread(target=self._process_document, args=(file,)).start()
+            if self.do_parent_document:
+                threading.Thread(target=self._process_document_parent_vector, args=(file,)).start()
 
             self.semaphore.acquire()
 
-            thread.start()
+            self._process_document_parent_vector(file)
 
     def save(self) -> None:
+        print(len(self.documents))
         np.save('data.npy', np.asarray(self.documents, dtype=object))
     
+    def _process_document_parent_vector(self, file) -> None:
+        _, extension = os.path.splitext(file)
+
+        loader = TextLoader(file)
+
+        match extension:
+            case ".pdf":
+                loader = UnstructuredPDFLoader(file)
+            case ".rst":
+                loader = UnstructuredRSTLoader(file)
+        
+        document_map = []
+        parent_docs = self.parent_splitter.split_documents(loader.load())
```

Comment:
It looks like theres a lot of shared code here, maybe you could combine the two functions together and add the parent doc functionality only if we're doing a parent document